x-pack/plugins/maps/server/maps_telemetry/maps_telemetry.ts

```diff
@@ -82,6 +94,112 @@ function getIndexPatternsWithGeoFieldCount(indexPatterns: IIndexPattern[]) {
   };
 }
 
+function getEMSLayerCount(layerLists: LayerDescriptor[][]): ILayerTypeCount[] {
+  return layerLists.map((layerList: LayerDescriptor[]) => {
+    const countsById = _(layerList).countBy((layer: LayerDescriptor) => {
+      const isEmsFile =
+        layer.sourceDescriptor !== null
+          ? layer.sourceDescriptor.type === SOURCE_TYPES.EMS_FILE
+          : false;
+      const id = (layer.sourceDescriptor as AbstractSourceDescriptor).id;
+      return isEmsFile && id ? id : 'false';
+    });
+
+    const emsCountsById = countsById.pickBy((val, key) => key !== 'false');
+    const layerTypeCount = emsCountsById.value();
+    return layerTypeCount as ILayerTypeCount;
+  }) as ILayerTypeCount[];
+}
+
+function isGeoshapeIndexPattern(
```

Comment:
Mind renaming to `isFieldGeoshape`, you are not testing if index pattern is geo shape but if field is geo shape.